packages/flutter_tools/lib/src/migrations/cmake_policy_migration.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/file_system.dart';
+import '../base/project_migrator.dart';
+import '../cmake_project.dart';
+
+// Migrate CMake's policy to CMP0135.
+// See https://github.com/flutter/flutter/issues/116866.
+class CmakePolicyMigration extends ProjectMigrator {
+  CmakePolicyMigration(CmakeBasedProject project, super.logger)
+    : _cmakeFile = project.cmakeFile;
+
+  final File _cmakeFile;
+
+  @override
+  void migrate() {
+    if (!_cmakeFile.existsSync()) {
+      logger.printTrace('CMake project not found, skipping cmake_policy() migration');
+      return;
+    }
+
+    final String originalProjectContents = _cmakeFile.readAsStringSync();
+    // Example:
+    //
+    // cmake_policy(SET CMP0063 NEW)
+
+    // Match the whole add_custom_command() and append VERBATIM unless it
+    // already exists.
+    final RegExp cmakePolicy = RegExp(
+      r'^cmake_policy\(SET CMP0063 NEW\)$',
+      multiLine: true,
+      dotAll: true,
+    );
+
+    String newProjectContents = originalProjectContents;
+
+    final Iterable<RegExpMatch> matches = cmakePolicy.allMatches(originalProjectContents);
+
+    for (final RegExpMatch match in matches) {
```

Comment:
In what scenarios would we have multiple matches? If that is an unexpected scenario, we shouldn't migrate the project.